### PR TITLE
Move file module

### PR DESCRIPTION
**Pull request by ThishaniLucas**
Fixing: https://github.com/ballerina-platform/ballerina-lang/issues/24803

### CODE REVIEW

File: file-ballerina/Ballerina.toml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+[project]
+org-name = "ballerina"
+version = "@toml.version@"
+
+[platform]
+target = "java8"
+
+    [[platform.libraries]]
+        artifactId = "file"
```
Comment (daneshk) [inline file-ballerina/Ballerina.toml:9]: Formatting issue

File: file-ballerina/src/file/tests/directory-listener-test.bal [hunk 2]
```diff
@@ -0,0 +1,72 @@
+import ballerina/test;
```
Comment (daneshk) [inline file-ballerina/src/file/tests/directory-listener-test.bal:1]: Shall we add license headers

File: file-ballerina/src/file/tests/file-test.bal [hunk 3]
```diff
@@ -0,0 +1,316 @@
+import ballerina/test;
```
Comment (daneshk) [inline file-ballerina/src/file/tests/file-test.bal:1]: Shall we add license headers

File: file-ballerina/src/file/tests/file-test.bal [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+function testRemoverecursiveTrue() {
+    var removeResult = remove(tmpdir+"/src-dir", true);
+    if (removeResult is error) {
+        test:assertFail("File not removed!");
+    }
+}
+
+@test:Config {}
+function testRemoveNonExistingFile() {
+    var removeResult = remove(tmpdir+"/no-file.txt");
```
Comment (daneshk) [inline file-ballerina/src/file/tests/file-test.bal:80]:
```suggestion
    var removeResult = remove(tmpdir + "/no-file.txt");
```

Comment (daneshk) [reply file-ballerina/src/file/tests/file-test.bal:80]: Please check all places

File: file-test-utils/src/main/java/org/ballerinalang/stdlib/file/testutils/TestUtil.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
+package org.ballerinalang.stdlib.file.testutils;
```
Comment (daneshk) [inline file-test-utils/src/main/java/org/ballerinalang/stdlib/file/testutils/TestUtil.java:1]: Add license headers

File: .gitignore [hunk 6]
```diff
@@ -21,3 +21,9 @@
 
 # virtual machine crash logs, see http://www.java.com/en/download/help/error_hotspot.xml
 hs_err_pid*
+
+# Ignore Gradle project-specific cache directory
+.gradle
+
+# Ignore Gradle build output directory
+build
```
Comment (BuddhiWathsala) [inline .gitignore:29]: Need to add directories such as target/ to the gitignore. Refer https://github.com/ballerina-platform/module-ballerina-socket/blob/81d949c548c490b61ac646c11bf6e1029ca954f9/.gitignore#L27. 

Basically, when you run the Gradle build there should not be any automatically generated unversioned files. Those unversioned files failed the release process when we release through the release plugin.

File: file-test-utils/build.gradle [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+ */
+
+plugins {
+    id 'java'
+}
+
+description = 'Ballerina - File Test Utils'
+
+dependencies {
+    compile group: 'org.ballerinalang', name: 'ballerina-lang', version: "${ballerinaLangVersion}"
```
Comment (BuddhiWathsala) [inline file-test-utils/build.gradle:25]: Do we need entire ballerina lang as a dependency?

Comment (ThishaniLucas) [reply file-test-utils/build.gradle]: Removed the lang dependency. Only the runtime is enough.

**Comment by daneshk** [comment]
Do test cases are running. We need to enable it in git action I think

**Comment by ThishaniLucas** [comment]
> Do test cases are running for PR. We need to enable it in git action I think

The tests are not skipped anywhere. Gradle build runs the test cases as well.

**Pull request merged**